app/controllers/application_controller.rb

```diff
@@ -45,6 +44,12 @@ def create_user_event(event_type, user = current_user)
 
   private
 
+  def redirect_on_timeout
+    return unless request.query_parameters[:timeout]
+    flash[:timeout] = t('notices.session_cleared')
+    redirect_to request.path
```

Comment:
fair point. I had considered that, then, suffering from false laziness, I saw that we had no existing usage of needing to persist parameters. Better to future-proof though.

Updated. PTAL.